### PR TITLE
Move textureData to itemStack BlockEntityTag

### PR DESCRIPTION
**Pull request by Nightenom**
- simplify passing textureData using vanilla mechanism 5869bc4 (applied in 0dd3dde)
  - made it non-breaking, but **please someone check with old mcol world it works as expected (chests/builders)**
- bump dev forge (tried to keep requirements same)
- add fix to BlockEntity#requestModelDataUpdate() so it works as expected [here](https://github.com/ldtteam/Domum-Ornamentum/commit/5869bc4ddf2fc2fda2ba6c34801d47fffa0ca8a9#diff-a063b0b5fe7a0b10cb923a83e205206e49ffbcb66ca077199eea96e31aaae9e1R72-R83) and here e6d6159

### CODE REVIEW

File: src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java [hunk 1]
```diff
@@ -41,19 +39,7 @@ public void updateTextureDataWith(final MaterialTextureData materialTextureData)
     @Override
     public @NotNull CompoundTag getUpdateTag()
     {
-        return this.saveWithId();
-    }
-
-    @Override
-    public void onDataPacket(final Connection net, final ClientboundBlockEntityDataPacket packet)
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java:51]: This is the client server sync, why remove it?

File: src/main/java/com/ldtteam/domumornamentum/block/vanilla/WallBlock.java [hunk 2]
```diff
@@ -118,19 +114,6 @@ public void fillItemCategory(final @NotNull NonNullList<ItemStack> items)
         items.addAll(fillItemGroupCache);
     }
 
-    @Override
-    public void setPlacedBy(
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/domumornamentum/block/vanilla/WallBlock.java:122]: In some cases you remove it and in some cases you change it. What is different there?

**Review by Raycoms — COMMENTED** [review]
Just some questions

File: src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java (hunk 1 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java:51]: override without any change

File: src/main/java/com/ldtteam/domumornamentum/block/vanilla/WallBlock.java (hunk 2 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/domumornamentum/block/vanilla/WallBlock.java:122]: i removed the be data setter it's handled by vanilla, however few methods like doors are needed cuz of other-half blocks, or some of them applies TYPE which i have no idea where it comes from so i havent touched it

File: src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java:51]: ah okay, this is the vanilla default. I always thought this was sort of empty by default =P

Comment (Nightenom) [reply src/main/java/com/ldtteam/domumornamentum/entity/block/MateriallyTexturedBlockEntity.java:51]: both are forge additions with default impl

**Comment by Nightenom** [comment]
Converted to draft due to data breaking nature

**Comment by Nightenom** [comment]
Made another PR with the type thing into blockState https://github.com/Nightenom/Domum-Ornamentum/pull/1

**Review by marchermans — APPROVED** [review]

**Pull request merged**